src/scripts/core/models/score-display.ts

```diff
@@ -4,63 +4,101 @@ import { Token } from "./token";
 import { eventEmitter } from "../../../event-emitter";
 import { ScoreMaths } from "../services/score-maths";
 
+export interface IScoreDisplayOptions {
+    titleString: string;
+    score: number;
+    onScoreChangeComplete: () => void;
+}
+
 export class ScoreDisplay extends Container {
 
-    private _score: number = 0;
-    private _scoreText: PixiText = new PixiText(`${this._score}`);
-    private _textContainer: Container = new Container();
-    private _trackedHeight = -1;
+    public score: number;
+    public scoreAsText: PixiText;
+    public textContainer: Container = new Container();
+    public onScoreChangeComplete: () => void;
```

Comment:
I couldn't see these variables being accessed outside of the class. Should they be public still?

Just as a general note I try to advocate keeping your class variables private/protected where possible. Providing a getter if something needs to access them, to limit class data being changed from outside accidentally.

Reply:
When I make the properties of ScoreDisplay private I get and error on GridScoreDisplay, which extends ScoreDisplay. Is there a better way for me to structure this?

Reply:
When extending a class and want to access the properties of the class you extend but not make them publicly accessible you want to use the 'protected' access modifier instead of public or private.

```
    protected score: number;
    protected scoreAsText: PixiText;
    protected textContainer: Container = new Container();
    protected onScoreChangeComplete: () => void;
```

Generally you only want to do this if you want extending classes to be able to access and modify these properties, additionally you can make them private and offer a public getter if you need access but don't wish for the extending class to be able to write to the property, there isn't really a right answer here but those are some of your options :)